tradeit-android-sdk/src/main/java/it/trade/android/sdk/model/TradeItLinkedBroker.java

```diff
@@ -40,9 +44,33 @@ public void onSuccessResponse(Response<TradeItAuthenticateResponse> response) {
                 TradeItSDK.getLinkedBrokerCache().cache(linkedBroker);
                 callback.onSuccess(linkedBrokerAccounts);
             }
+
+            @Override
+            public void onErrorResponse(TradeItErrorResult errorResult) {
+                linkedBroker.error = errorResult;
+                callback.onError(errorResult);
+            }
         });
     }
 
+    public void authenticateIfNeeded(final TradeItCallbackWithSecurityQuestion<List<TradeItLinkedBrokerAccount>> callback) {
+        if (this.error != null && this.error.requiresAuthentication()) {
+            this.authenticate(callback);
+        } else if (this.error != null && this.error.requiresRelink()) {
```

Comment:
What about 101/301? Shouldn't those all also be
```java
callback.onError(this.error);
```

I think we need to add them to the AndroidAPI also because they are not on the `TradeItErrorCode` enum.